src/engine/Service.cpp

```diff
@@ -38,7 +38,7 @@ std::string Service::getCacheKeyImpl() const {
   }
   os << parsedServiceClause_.serviceIri_.toStringRepresentation() << " {\n"
      << parsedServiceClause_.prologue_ << "\n"
-     << parsedServiceClause_.graphPatternAsString_ << "\n";
+     << getGraphPattern() << "\n";
```

Comment:
The cache Key of the SERVICE is wrong.
It should be a random number, to never cache the result.
Either add a TODO here, or fix it directly.

Reply:
In particular there are two reasons that it is wrong:
1. It can be the remote endpoint has been update since the last time.
2. The cache key doesn't contain the sibling when the key is initialized, but then suddenly changes, that also doesn't work properly.